spec/features/01_next_up_spec.rb

```diff
@@ -0,0 +1,179 @@
+require "rails_helper"
+
+describe "The Next Up section" do
+  it "has a div element with the class \"next_up\"", points: 1 do
+    visit("/user_sign_in")
+    user_jacob = User.new
+    user_jacob.email = "jacob_#{rand(100)}@example.com"
+    user_jacob.password = "password"
+    user_jacob.save
+
+    visit "/user_sign_in"
+    
+    within(:css, "form") do
+      fill_in "Email", with: user_jacob.email
+      fill_in "Password", with: user_jacob.password
+      find("button", :text => /Sign in/i ).click
+    end
+
+    visit("/")
+
+    expect(page).to have_css("div.next_up"),
+      "Expected to find a <div> element with the a class attribute containing 'next_up', but didn't find one."
+  end
+end
+
+describe "The Next Up section" do
+  it "has the color of all text styled as darkorange",js: true, points: 1 do
+    visit("/user_sign_in")
+    user_jacob = User.new
+    user_jacob.email = "jacob_#{rand(100)}@example.com"
+    user_jacob.password = "password"
+    user_jacob.save
+
+    visit "/user_sign_in"
+    
+    within(:css, "form") do
+      fill_in "Email", with: user_jacob.email
+      fill_in "Password", with: user_jacob.password
+      find("button", :text => /Sign in/i ).click
+    end
+
+    visit("/")
+    next_up_div = find("div.next_up")
+
+    # next_up_div.assert_matches_style( 'color' => 'darkorange')
+    expect(next_up_div).to have_color("orange")
+
+  end
+end
+
+describe "The Next Up section" do
+  it "has a form that updates a todo item, and moves it to the \"In Progress\" section", points: 1 do
+    visit("/user_sign_in")
+    user_jacob = User.new
+    user_jacob.email = "jacob_#{rand(100)}@example.com"
+    user_jacob.password = "password"
+    user_jacob.save
+
+    visit "/user_sign_in"
+
+    within(:css, "form") do
+      fill_in "Email", with: user_jacob.email
+      fill_in "Password", with: user_jacob.password
+      find("button", :text => /Sign in/i ).click
+    end
+
+    visit("/")
+
+    within(:css, "form") do
+      fill_in "Describe the task:", with: "Work on Todo List app"
+      find("button", :text => /Add to Next Up/i ).click
+    end
+
+    within(:css, "div.next_up") do
+      within(:css, "form") do
```

Comment:
In theory they could make a link that, when clicked, does the job (that's how I did it initially).

Can we find something (anything) with the text /Move to In Progress/i and click it? IIRC we moved away from `click_on` because it was hard to do case-insensitivity.  But maybe allowing freedom of choice in implementation outweighs that, especially at this stage (they should be used to matching the copy exactly by now)?

Reply:
Ah, you're right. `find` is the more flexible method to use for clicking buttons with regex. It also provides a slightly clearer error message on failures.